agent/lib/src/external_api/model.rs

```diff
@@ -0,0 +1,139 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub struct FileModel {
+    pub filename: String,
+    pub content: String,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct CodeEntry {
+    pub files: Vec<FileModel>,
+    pub script: Vec<String>, // All commands to execute at startup
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Type {
+    Status,
+    Request,
+    Response,
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Code {
+    Run,
+    Ok,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct StatusMessage {
+    pub r#type: Type,
+    pub code: Code,
+}
+
+impl StatusMessage {
+    pub fn new() -> StatusMessage {
+        StatusMessage {
+            r#type: Type::Status,
+            code: Code::Ok,
+        }
+    }
+}
+
+impl Default for StatusMessage {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct ResponseStep {
+    pub command: String,
+    pub result: i32,
+    pub stdout: String,
+    pub stderr: String,
+    pub enable_output: bool,
+}
+
+impl ResponseStep {
+    pub fn new(
+        command: String,
+        result: i32,
+        stdout: String,
+        stderr: String,
+        enable_output: bool,
+    ) -> ResponseStep {
+        ResponseStep {
+            command,
+            result,
+            stdout,
+            stderr,
+            enable_output,
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct ResponseData {
+    pub id: String,
+    pub steps: Vec<ResponseStep>,
+}
+
+impl ResponseData {
+    pub fn new(id: String, steps: Vec<ResponseStep>) -> ResponseData {
+        ResponseData { id, steps }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct ResponseMessage {
+    pub r#type: Type,
+    pub code: Code,
+    pub data: ResponseData,
+}
+
+impl ResponseMessage {
+    pub fn new(data: ResponseData) -> ResponseMessage {
+        ResponseMessage {
+            r#type: Type::Response,
+            code: Code::Run,
+            data,
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
+pub struct RequestStep {
+    pub command: String,
+    pub enable_output: bool,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct RequestData {
+    pub id: String,
+    pub files: Vec<FileModel>,
+    pub steps: Vec<RequestStep>,
+}
+
+impl RequestData {
+    pub fn new(id: String, files: Vec<FileModel>, steps: Vec<RequestStep>) -> RequestData {
+        RequestData { id, files, steps }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct RequestMessage {
+    pub r#type: Type,
+    pub code: Code,
+    pub data: RequestData,
+}
+
+impl RequestMessage {
+    pub fn new(data: RequestData) -> RequestMessage {
+        RequestMessage {
+            r#type: Type::Request,
+            code: Code::Run,
```

Comment:
Is code here describing a type of request (e.g. `Run`,  or `Kill` or `Abort`, etc)?

Reply:
Yes. Though we only have ``Run`` to indicate that the message is about running code, and ``Ok`` to indicate that the agent is up and running.

We do not have an ``Abort`` or ``Kill`` code as we determind that we could simply kill the VM in the [issue thread](https://github.com/virt-do/lambdo/issues/5#issuecomment-1475410352)